design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,711 @@
+- Duplicating this logic gives us a 1:1 map between the “interface” field on the networks-status annotations (net1, net2, etc..) and the network name.
+
+##### Example
+The following VMI has 1 bridge and 2 SR-IOV secondary networks:
+
+```yaml
+spec:
+  domain:
+    devices:
+      Interfaces:
+      - name: bridge-primary-mac
+        macAddress: aa:bb:cc:dd:ee:00
+        sriov: {}
+      - name: sriovnet-vlan100-secondary-mac
+        macAddress: aa:bb:cc:dd:ee:01
+        sriov: {}
+      - name: sriovnet-vlan100-third-mac
+        macAddress: aa:bb:cc:dd:ee:02
+        sriov: {}
+      ...
+  Networks:
+  - multus:
+        networkName: default/bridge-network
+    name: bridge-primary-mac
+  - multus:
+        networkName: default/sriov-network-vlan100
+    name: sriovnet-vlan100-secondary-mac
+  - multus:
+        networkName: default/sriov-network-vlan100
+    name: sriovnet-vlan100-third-mac
+  ...
+```
+
+We will map the “interface” field to the network-name as follows:
+- bridge-primary-mac ⇒ net1
+- sriovnet-vlan100-secondary-mac ⇒ net2
+- sriovnet-vlan100-third-mac ⇒ net3
+
+
+With this mapping we can easily retrieve the PCI address, as the one with the corresponding “interface” name:
+```yaml
+k8s.v1.cni.cncf.io/network-status: |-
+    [{
+          "name": "kindnet",
+          "interface": "eth0",
+          "ips": [
+              "10.244.1.9"
+          ],
+          "mac": "3a:7e:42:fa:37:c6",
+          "default": true,
+          "dns": {}
+      },{
+          "name": "default/bridge-network",
+          "interface": "net1",
+          "mac": "aa:bb:cc:dd:ee:00",
+          "dns": {}
+      },{
+          "name": "default/sriov-network-vlan100",
+          "interface": "net2",
+          "mac": "aa:bb:cc:dd:ee:01",
+          "dns": {},
+          "device-info": {
+              "type": "pci",
+              "version": "1.0.0",
+              "pci": {
+                  "pci-address": "0000:65:00.2"
+              }
+      },{
+          "name": "default/sriov-network-vlan100",
+          "interface": "net3",
+          "mac": "aa:bb:cc:dd:ee:02",
+          "dns": {},
+          "device-info": {
+              "type": "pci",
+              "version": "1.0.0",
+              "pci": {
+                  "pci-address": "0000:65:00.3"
+              }
+          }
+      }]
+```
+
+Since we’re only interested in the SRIOV networks, the network-name-to-pci-address mapping will be:
+- sriovnet-vlan100-secondary-mac ⇒ 0000:65:00.2
+- sriovnet-vlan100-third-mac ⇒ 0000:65:00.3
+
+> **_Note_**:
+> We are relying on the fact that the “interface” naming convention in [[2]](https://github.com/kubevirt/kubevirt/blob/c4b6ae63c5a7f642ab86b0755dabca3b814ecb39/pkg/virt-controller/services/template.go#L1692) will remain the same.
+> To ensure that, we plan cover it with e2e test in case the naming logic is changed without doing the same in the SRIOV device creation.
```

Comment:
```suggestion
> To ensure that, we plan move this logic to a new package that will be used for all scenarios related to this annotations. Moreover, this will be covered in e2e test in case the naming logic is changed without using the new package.
```

Reply:
Sounds good, done.